R/FilterStates.R

```diff
@@ -492,6 +492,11 @@ FilterStates <- R6::R6Class( # nolint
           NULL
         }
       )
+    },
+
+    finalize = function() {
+      .finalize_observers(self, private) # Remove all observers
+      .finalize_state_list(self, private) # Remove all FilterState objects
```

Comment:
Easier to comprehend if this util function is not called here 
```suggestion
     private$state_list_empty(force = TRUE)
     isolate(private$state_list(NULL))
```

Reply:
Done! Good point